lib/common/aws_iot_shared.ts

```diff
@@ -157,36 +155,35 @@ export interface MqttConnectCustomAuthConfig {
      * The digital signature of the token value in the {@link tokenValue} property.  The signature must be based on
      * the private key associated with the custom authorizer.  The signature must be base64 encoded.
      *
-     * Required if the custom authorizer has signing enabled.  It is strongly suggested to URL-encode this value; the
-     * SDK will not do so for you.
+     * Required if the custom authorizer has signing enabled.
      */
     tokenSignature?: string;
 };
 
 /** @internal */
-export function canonicalizeCustomAuthTokenSignature(signature?: string) : string | undefined {
-    if (signature === undefined || signature == null) {
+export function canonicalizeCustomAuthEncoding(value?: string) : string | undefined {
+    if (value === undefined || value == null) {
         return undefined;
     }
 
-    let hasPercent = signature.indexOf("%") != -1;
+    let hasPercent = value.indexOf("%") != -1;
     if (hasPercent) {
-        return signature;
+        return value;
     } else {
-        return encodeURIComponent(signature);
+        return encodeURIComponent(value);
     }
 }
 
 /** @internal */
 export function canonicalizeCustomAuthConfig(config: MqttConnectCustomAuthConfig) : MqttConnectCustomAuthConfig {
     let processedConfig : MqttConnectCustomAuthConfig = {};
 
-    utils.set_defined_property(processedConfig, "authorizerName", config.authorizerName);
+    utils.set_defined_property(processedConfig, "authorizerName", canonicalizeCustomAuthEncoding(config.authorizerName));
```

Comment:
Question: regards to the authorizerName. As the canonicalizeCustomAuthEncoding  only checks for `%` , is it possible that username contains `%`? I thought we use `%`  to validate the signature because the signature is base64 encoded and should never include a `%`. Is that also true to authorizerName?

Reply:
AWS documentation suggests uri-encoding all three of: token key name, token value signature, and authorizer name.  If an authorizer name did contain a % naturally (unsure if it's allowed) then it would have to be encoded as "%25" which is exactly what we do here.

Reply:
I think our `canonicalizeCustomAuthEncoding` would return the original value instead of encoding it if it contains a `%`. If `%` is allowed for authorizer name, it seems that we would not encode it here with `canonicalizeCustomAuthEncoding`.

Reply:
Hmm you're right.  Need to rethink this.